src/benchmarklib/benchmark_sql_executor.cpp

```diff
@@ -41,16 +46,30 @@ std::pair<SQLPipelineStatus, std::shared_ptr<const Table>> BenchmarkSQLExecutor:
   return {pipeline_status, result_table};
 }
 
+BenchmarkSQLExecutor::~BenchmarkSQLExecutor() {
+  // If the benchmark item did not call commit or rollback, we need to close the sqlite transaction so that the
+  // next item can start a new one:
+  if (_sqlite_transaction_open) _sqlite_wrapper->raw_execute_query("COMMIT TRANSACTION");
```

Comment:
> If the benchmark item does not explicitly manage the transaction life time by calling commit or rollback, we automatically commit the sqlite transaction so that the next item can start a new one

better?

Reply:
Yes. But why isn't this behavior ("..so that next item can start a new one") possible with a rollback?